ShopAppTests/App/Main/Views/Cells/CollectionView/ProductOptionCollectionViewCellSpec.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  ProductOptionCollectionViewCellSpec.swift
+//  ShopAppTests
+//
+//  Created by Evgeniy Antonov on 3/21/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import Quick
+import Nimble
+import ShopApp_Gateway
+
+@testable import ShopApp
+
+class ProductOptionCollectionViewCellSpec: QuickSpec {
+    override func spec() {
+        var cell: ProductOptionCollectionViewCell!
+        var collectionView: UICollectionView!
+        var optionTitleLabel: UILabel!
+        
+        beforeEach {
+            let provider = ProductOptionsCollectionCellProvider()
+            
+            collectionView = UICollectionView(frame: CGRect.zero, collectionViewLayout: UICollectionViewFlowLayout())
+            collectionView.registerNibForCell(ProductOptionCollectionViewCell.self)
+            
+            collectionView.dataSource = provider
+            collectionView.delegate = provider
+            
+            provider.values = ["Value 1"]
+            provider.selectedValue = "Value 1"
+            collectionView.reloadData()
+            
+            let indexPath = IndexPath(row: 0, section: 0)
+            let dequeuedCell: ProductOptionCollectionViewCell = collectionView.dequeueReusableCellForIndexPath(indexPath)
+            cell = dequeuedCell
+            
+            optionTitleLabel = self.findView(withAccessibilityLabel: "label", in: cell) as! UILabel
+        }
+        
+        describe("when cell configured") {
+            var text: String!
+            var selected: Bool!
+            
+            beforeEach {
+                text = "Text"
+            }
+            
+            context("if option selected") {
+                beforeEach {
+                    selected = true
+                    cell.configure(with: text, selected: selected)
+                }
+                
+                it("should have correct title") {
+                    expect(optionTitleLabel.text) == text
+                }
+                
+                it("should have correct text color") {
+                    expect(optionTitleLabel.textColor) == .white
+                }
+                
+                it("should have correct background color") {
+                    expect(cell.backgroundColor) == .black
+                }
+                
+                it("should have correct corner radius") {
```

Comment:
Take out this block before `context` blocks because it isn't different for both `contexts`:

```
it("should have correct corner radius") {
  cell.configure(with: "", selected: true)

  expect(cell.layer.cornerRadius) == cell.frame.size.height / 2
}
```

Reply:
This property would calculated after calling 'configure cell', which called from 'beforeEach' of context. So we can create separated context for checking only corner radius property. Or check it in one or both exist contexts